pkg/odo/cli/catalog/list/components.go

```diff
@@ -57,10 +57,14 @@ func (o *ListComponentsOptions) Complete(name string, cmd *cobra.Command, args [
 	}
 
 	if experimental.IsExperimentalModeEnabled() {
-		o.catalogDevfileList, err = catalog.ListDevfileComponents()
+		o.catalogDevfileList, err = catalog.ListDevfileComponents("")
```

Comment:
This argument is for specifying registry name and ideally this argument should be optional so that we list devfile components from all registries without passing in specific registry name, but Go doesn't support optional argument so here we pass empty string `""` means there is no specific registry name.

Reply:
Yes, Go does't support optional arguments for a reason. We shouldn't have to pass in `""` into the function. I'd advise that we make this into a separate function with no passing in and create another function called `FilterDevfileComponents` so you can pass in a string. That's just IMO.

Reply:
Actually they share the same function body only difference is the optional function argument, I would say creating another function would cause duplication and function inconsistency. `ioutil.TempFile()` uses the same way to handle optional argument case for default filepath with proper comment, I think we can do the same thing.